kb/development/script-coding-standards.md

```diff
@@ -0,0 +1,102 @@
+# Mbed TLS script coding standards
+
+## Choice of scripting language
+
+The preferred language for scripts in the Mbed TLS project is Python. Use Python for new scripts that are committed to the project unless there's a good reason.
+
+Some reasons not to use Python:
+
+* A script that only applies to Windows and should ideally work out of the box can be written in cmd.
+* Some CI code in the `mbedtls-test` repository is written in Groovy, for its Jenkins integration.
+* A script only intended for the Mbed TLS developers, and whose job is mostly to run other programs, can be written in sh.
+* If you're extending existing scripts, then obviously keep using the same language.
+* Throwaway scripts that do ad hoc automated refactoring (e.g. renaming identifiers) are often in Perl, thanks to its convenient text manipulation primitives.
+
+### Language versions
+
+Check [`README.md`](https://github.com/Mbed-TLS/mbedtls/blob/development/README.md#tool-versions) for the current minimum supported version of Python and other tools. Note that in addition to the public commitments, at any given time, there may be additional constraints due to the platforms on which we run automated tests. As of September 2022, Python scripts must be compatible with Python 3.5.
+
+For scripts that need to be backported, note that [long-time support branches](https://github.com/Mbed-TLS/mbedtls/blob/development/BRANCHES.md#long-time-support-branches) may support older versions of Python and other tools than the `development` branch.
+
+### Script naming conventions
```

Comment:
Nit: I don't think "Script naming conventions", "Invocation conventions" and "File structure" should be subsections of "Choice of scripting language", logically.

I'd prefer to have all four of them on the same level of heading.